### ISSUE TITLE
Wraith climbed stairs whilst being stuck

### ISSUE THREAD

**Issue by flend** [flend/BrogueCE#1]
RapidBrogue v1.0.0, Turn 2756, seed 271503552
http://brogue.roguelikelike.com/#viewRecording/RAPIDBROGUEV100-60cf7d4a9889c83a28f36223
![image](https://user-images.githubusercontent.com/3458348/122716652-72d32400-d262-11eb-91d1-8aa44abd2cd4.png)

**Comment by flend** [comment]
From Lordans:
fwiw the first weird thing happens at 2745 when, if you hover over the wraith, you get the message "The wraith is entangled, burning, in billowing flames"
Lordans: I suspect it has something to do with the fact that I was hasted, and wraiths move quickly, as well as us both being on fire in spiderwebs
it looked like it didn't interfere with movement but allowed me to get a "dangle helplessly" hit

**Comment by flend** [comment]
Save game for RB 1.0.0: [webbrogue-recording-60cf7d4a9889c83a28f36223.zip](https://github.com/flend/BrogueCE/files/6691271/webbrogue-recording-60cf7d4a9889c83a28f36223.zip)

**Comment by flend** [comment]
The wraith appears to be able to move normally for multiple turns but has the Stuck status (which does not gradually go away)

**Comment by abedolinger** [comment]
Isn't it because it's burning?  When I'm on fire in regular Brogue, I can move through nets/spiderwebs at 1 space per turn.

**Comment by MattWeiner** [comment]
I don't think that being burning explains it entirely. If you start at 2729, the wraith is on fire, and next turn gets the message "The wraith is stuck fast in a spiderweb!" but it doesn't have the Stuck status and the hover message is "The wraith is charging, burning, through billowing flames." Then on 2731 when I catch fire that removes the Stuck status.

On 2732 the wisp gets Stuck status and the hover message "The wisp is entangled in billowing flames." This seems like it maybe shouldn't happen as it allows me to get free hits in.

Then on 2737 when the spider launches webs neither the wraith nor I get Stuck as the webs on our squares seem to catch fire instantly (FWIW they were already fire squares so maybe webs just don't generate on them).

On 2745, after I haste myself, the wraith gets Stuck when the spider launches webs with "The wraith is entangled, burning, in billowing flames." (The wraith was already burning and I wasn't, on squares that weren't on fire when the spider launched the webs.) Then we proceed to be sometimes entangled in billowing flames but free to move, till we get the really weird status on the wraith when it climbs the stairs.
